### PR TITLE
8301991: Convert l10n properties resource bundles to UTF-8 native

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
JDK .properties files still use ISO-8859-1 encoding with escape sequences. It would improve readability to see the native characters instead of escape sequences (especially for the L10n process). The majority of files changed are localized resource files.

This change converts the Unicode escape sequences in the JDK .properties files (both in src and test) to UTF-8 native characters. Additionally, the build logic is adjusted to read the .properties files in UTF-8 while generating the ListResourceBundle files.

The only escape sequence not converted was `\u0020` as this is used to denote intentional trailing white space. (E.g. `key=This is the value:\u0020`)

The conversion was done using native2ascii with options `-reverse -encoding UTF-8`.

If this PR is integrated, the IDE default encoding for .properties files need to be updated to UTF-8. (IntelliJ IDEA locks .properties files as ISO-8859-1 unless manually changed).

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301991](https://bugs.openjdk.org/browse/JDK-8301991): Convert l10n properties resource bundles to UTF-8 native (**Enhancement** - P4)

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15694/head:pull/15694` \
`$ git checkout pull/15694`

Update a local copy of the PR: \
`$ git checkout pull/15694` \
`$ git pull https://git.openjdk.org/jdk.git pull/15694/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15694`

View PR using the GUI difftool: \
`$ git pr show -t 15694`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15694.diff">https://git.openjdk.org/jdk/pull/15694.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15694#issuecomment-1716559612)

### CODE REVIEW

File: make/jdk/src/classes/build/tools/compileproperties/CompileProperties.java [hunk 1]
```diff
@@ -220,8 +222,11 @@ private static boolean createFile(String propertiesPath, String outputPath,
             System.out.println("parsing: " + propertiesPath);
         }
         Properties p = new Properties();
-        try {
-            p.load(new FileInputStream(propertiesPath));
+        try (FileInputStream input = new FileInputStream(propertiesPath);
+             // Read in JDK .properties files in UTF-8
+             InputStreamReader streamReader = new InputStreamReader(input, StandardCharsets.UTF_8)
```
Comment (liach) [inline make/jdk/src/classes/build/tools/compileproperties/CompileProperties.java:227]: Can we just uses `Files.newBufferedReader(Path.of(propertiesPath))` instead?

**Review by naotoj — APPROVED** [review]
Looks good to me, although I did not look at each l10n file, but sampled some. Thanks for tackling this conversion.

**Comment by justin-curtis-lu** [comment]
> Looks good to me, although I did not look at each l10n file, but sampled some. Thanks for tackling this conversion.

Thanks for the review; (In addition to testing), I ran a script to verify only white space escape sequences exist in JDK .properties files. (Excluding escape sequences in test files that should remain as is for the purpose of the test)

File: src/jdk.jartool/share/classes/sun/tools/jar/resources/jar_pt_BR.properties [hunk 2]
```diff
@@ -23,103 +23,103 @@
...
 
 main.usage.summary=Uso: jar [OPTION...] [ [--release VERSION] [-C dir] files] ...
-main.usage.summary.try=Tente `jar --ajuda' para obter mais informa\u00E7\u00F5es.
+main.usage.summary.try=Tente `jar --ajuda' para obter mais informações.
```
Comment (pavelrappo) [inline src/jdk.jartool/share/classes/sun/tools/jar/resources/jar_pt_BR.properties:95]: I was looking for something unrelated in properties files, and found this. It is surprising to see an option name being localised; it must be a bug.

Comment (naotoj) [reply src/jdk.jartool/share/classes/sun/tools/jar/resources/jar_pt_BR.properties:95]: Good catch, Pavel. It is indeed a bug. This type of overtranslation l10n bug happens all the time, and hard to catch.

**Comment by magicus** [comment]
Continuing the discussion that was started at a predecessor to this PR, https://github.com/openjdk/jdk/pull/12726#discussion_r2035582242. At least one incorrect conversion has been found in this PR. It might be worthwhile to double- and triple-check all the other conversions as well.

**Comment by magicus** [comment]
As part of https://bugs.openjdk.org/browse/JDK-8301971 I am trying various ways of detecting files without UTF-8 encoding, but it is still a bit of hit and miss, since there are no surefire way of telling which encoding a file has, only heuristics. So finding and following up potential sources of error is important.

**Comment by eirbjo** [comment]
FWIW, I checked out the revision of the commit previous to this change and found the following:

```
% git checkout b55e418a077791b39992042411cde97f68dc39fe^ 
% find src -name "*.properties" | xargs file | grep -v ASCII
src/java.xml/share/classes/com/sun/org/apache/xml/internal/serializer/Encodings.properties:                                                                                                              ISO-8859 text
src/java.xml.crypto/share/classes/com/sun/org/apache/xml/internal/security/resource/xmlsecurity_de.properties:                                                                                           Unicode text, UTF-8 text, with very long lines (322)
```

Which indicates that that this is the only non-ASCII, non-UTF-8 file. So we may be lucky.

**Comment by magicus** [comment]
Thanks for checking!

**Pull request closed without merging**